client/packages/host/src/Admin/WebAppLogFileModal.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from 'react';
+import { useDialog, useWebClient } from '@common/hooks';
+import { useTranslation } from '@common/intl';
+import {
+  BasicSpinner,
+  DialogButton,
+  DropdownMenu,
+  DropdownMenuItem,
+  LoadingButton,
+  Typography,
+} from '@common/components';
+import { useLog } from '@openmsupply-client/system';
+import { Box, CopyIcon, SaveIcon } from 'packages/common/src';
+import { LogTextDisplay } from './LogTextDisplay';
+
+export const LogDisplay = ({
+  fileName,
+  setLogContent,
+}: {
+  fileName: string;
+  setLogContent: (content: string[]) => void;
+}) => {
+  const { data, isLoading } = useLog.document.logContentsByFileName(fileName);
+
+  if (data?.fileContent !== undefined && data?.fileContent !== null) {
+    setLogContent(data?.fileContent);
+  }
+
+  if (isLoading) {
+    return (
+      <>
+        <BasicSpinner></BasicSpinner>
+      </>
+    );
+  } else {
+    return (
+      <>
+        {Array.isArray(data?.fileContent) && data?.fileContent != undefined ? (
+          <LogTextDisplay logText={data?.fileContent}></LogTextDisplay>
+        ) : null}
+      </>
+    );
+  }
+};
+
+export const WebAppLogFileModal = ({
+  isOpen,
+  onClose,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+}) => {
+  const t = useTranslation('common');
+  const [logToRender, setLogToRender] = useState('');
+  const [logContent, setLogContent] = useState<string[]>([]);
+  const { Modal } = useDialog({ isOpen });
+  const { saveFile } = useWebClient();
+  const [isSaving, setIsSaving] = useState(false);
+
+  const {
+    data: logFiles,
+    isError,
+    isLoading: logListLoading,
+  } = useLog.document.listFileNames();
+
+  const saveLog = async () => {
+    setIsSaving(true);
+    await saveFile({
+      content: logContent.toString(),
+      filename: `${logToRender}-exported-log.txt`,
```

Comment:
```suggestion
      filename: `${logToRender}`,
```
I think it's fine to get rid of this?